src/components/date-picker/DatePicker.tsx

```diff
@@ -440,17 +610,33 @@ class DatePicker extends React.Component<DatePickerProps> {
 
         const resinTargetAttr = resinTarget ? { [RESIN_TAG_TARGET]: resinTarget } : {};
 
-        const valueAttr = isTextInputAllowed
-            ? { defaultValue: this.formatDisplay(value) }
-            : { value: this.formatDisplay(value) };
+        let valueAttr;
+        if (isAccessible) {
+            valueAttr = { defaultValue: this.formatDisplayDateType(value) };
+        } else if (isTextInputAllowed) {
+            valueAttr = { defaultValue: this.formatDisplay(value) };
+        } else {
+            valueAttr = { value: this.formatDisplay(value) };
+        }
+        let onChangeAttr;
+        if (isAccessible && this.canUseDateInputType) {
+            onChangeAttr = { onChange: this.handleOnChange };
+        } else if (isTextInputAllowed || (isAccessible && !this.canUseDateInputType)) {
+            onChangeAttr = {};
+        } else {
+            onChangeAttr = { onChange: noop };
+        }
+        let patternAttr = {};
+        if (isAccessible && !this.canUseDateInputType) {
+            patternAttr = { pattern: '\\d{4}-\\d{2}-\\d{2}' };
+        }
+        // "name" prop is required for pattern validation to be surfaced on form submit. See components/form-elements/form/Form.js
+        let nameAttr = {};
+        if (isAccessible && !this.canUseDateInputType) {
+            nameAttr = { name };
+        }
```

Comment:
Seems like these two can be combined, might be worth seeing if this block of code can be simplified further.

Reply:
I'll work on cleaning this part up